### PR TITLE
[GLUTEN-6304][CH]Support array_join

### PR DESCRIPTION
**Pull request by KevinyhZou**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#6304)

## How was this patch tested?

test by spark ut

### CODE REVIEW

File: cpp-ch/local-engine/Parser/SerializedPlanParser.h [hunk 1]
```diff
@@ -177,7 +177,8 @@ static const std::map<std::string, std::string> SCALAR_FUNCTIONS
        {"array", "array"},
        {"shuffle", "arrayShuffle"},
        {"range", "range"}, /// dummy mapping
-        {"flatten", "sparkArrayFlatten"},
+       {"flatten", "sparkArrayFlatten"},
+       {"array_join", "sparkArrayJoin"},
```
Comment (lgbo-ustc) [inline cpp-ch/local-engine/Parser/SerializedPlanParser.h:181]: Please don't add new functions by this map

Comment (lgbo-ustc) [reply cpp-ch/local-engine/Parser/SerializedPlanParser.h:181]: Use the `FunctionParser`

Comment (KevinyhZou) [reply cpp-ch/local-engine/Parser/SerializedPlanParser.h:181]: array_join not has function to map in clickhouse

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**